sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -459,12 +548,284 @@ edges.
+  the node.
+
+* Throws with error code `invalid` if `param` is not a buffer accessor.
+
+* Throws with error code `invalid` if the type of the node is not a kernel
+  execution.
+
+|
+[source,c++]
+----
+template <typename T>
+void register_dynamic_parameter(
+    T param,
+    dynamic_parameter<dynamic_parameter_type::scalar> dynamicParam) const;
+----
+|Associate a scalar dynamic parameter with a scalar kernel parameter of this node.
+All scalar kernel arguments matching `param` are bound to `dynamicParam`.
+
+See <<executable-graph-update, Executable Graph Update>> for more information
+about updating node parameters.
+
+Parameters:
+
+* `param` - Scalar used as a node argument.
+* `dynamicParam` - Dynamic parameter which is to be associated with the kernel
+   arguments matching `param`.
+
+Exceptions:
+
+* Throws with error code `invalid` if `param` does not match any arguments in
+  the node.
+
+* Throws with error code `invalid` if `param` does not match the specified on
+  creation of `dynamicParam`.
+
+* Throws with error code `invalid` if the type of the node is not a kernel
+  execution.
+
+|
+[source,c++]
+----
+template <dynamic_parameter_type DynParamType>
+void register_dynamic_parameter(
+    int argIndex,
+    dynamic_parameter<DynParamType> dynamicParam) const;
+----
+|Associate a dynamic parameter with a kernel parameter of this node via the
+argument index. Although argument ordering is not defined for lambda captures,
+the ordering is reliable for arguments set using `handler::set_arg()` or
+`handler::set_args()`.
+
+See <<executable-graph-update, Executable Graph Update>> for more information
+about updating node parameters.
+
+Parameters:
+
+* `argIndex` - Kernel argument index starting from zero to associate
+  `dynamicParam` with.
+* `dynamicParam` - Dynamic parameter which is associated with the kernel
+   argument at index `argIndex`.
+
+Exceptions:
+
+* Throws with error code `invalid` if `argIndex` is not a valid argument index
+  for the kernel.
+
+* Throws with error code `invalid` if the type of `dynamicParam` does not
+  correspond to the type of the node at `argIndex`.
+
+* Throws with error code `invalid` if the type of the node is not a kernel
+  execution.
+
+|
+[source,c++]
+----
+node_type get_type() const;
+----
+|Returns a value representing the type of command this node represents.
+
+|
+[source,c++]
+----
+std::vector<node> get_dependencies() const;
+----
+|Returns a list of the nodes which this node depends on.
+
+|
+[source,c++]
+----
+std::vector<node> get_dependent_nodes() const;
+----
+|Returns a list of the nodes which depend on this node.
+
+|
+[source,c++]
+----
+static node get_node_from_event(event nodeEvent);
+----
+|Finds the node associated with an event created from a submission to a queue
+ in the recording state.
+
+Parameters:
+
+* `nodeEvent` - Event returned from a submission to a queue in the recording
+  state.
+
+Returns: Graph node associated to `nodeEvent`.
```

Comment:
shall we add something to make it clear, something like:  the completion of nodeEvent means the node is finished.

Reply:
That's not the current semantics, you can't query event status at a node level. From the section on Event Limitations:

> Calling `event::get_info<info::event::command_execution_status>()` or  `event::get_profiling_info()` will throw synchronously with error code `invalid`.

Reply:
my idea was to describe the relation between `node` and `nodeEvent` in a more straight-forward way:  the submission of node returns nodeEvent.